File: LeetCode/Easy/Valid Parentheses/Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        open_parentheses = ["(","[","{"]
        close_parentheses = [")","]","}"]
        
        stack = []
        
        for i in s:
            if i in open_parentheses:
                stack.append(close_parentheses[open_parentheses.index(i)])
            else:
                if len(stack) != 0 and stack[-1] == i:
                    stack.pop()
                else:
                    return False
        if len(stack) != 0:
            return False
        return True

File: LeetCode/Easy/Valid Parentheses/Valid Parentheses.go
func isValid(s string) bool {
    
    if len(s) == 0 || len(s)%2 !=0{
        return false
    }
    dict := map[rune]rune{
        '(':')',
        '[':']',
        '{':'}',
    }
    var stack []rune
    
    for _, val := range s{
        if _, exists := dict[val]; exists{
            stack = append(stack, dict[val])
        }else if len(stack) == 0 || stack[len(stack)-1] != val{
            return false
        }else{
        stack = stack[:len(stack)-1]
        }
    }
    if len(stack) !=0{
        return false
    }
    return true
}
